compiler/src/formatting/fmt.re

```diff
@@ -0,0 +1,3348 @@
+                         next.pexp_loc,
+                       ),
+                ~trail=
+                  prev =>
+                    trailing_comma
+                    ++ print_comment_range(
+                         ~lead=space,
+                         ~block_end=true,
+                         prev.pexp_loc,
+                         enclosing_end_location(expr.pexp_loc),
+                       ),
+                ~f=print_expression,
+                exprs,
+              ),
+            )
+          | PExpConstrRecord(exprs) =>
+            braces(
+              concat_map(
+                ~lead=
+                  ((next_ident, _)) =>
+                    print_comment_range(
+                      ~block_start=true,
+                      ~trail=space,
+                      ident_loc,
+                      next_ident.loc,
+                    ),
+                ~sep=
+                  ((_, prev), (next, _)) =>
+                    comma
+                    ++ print_comment_range(
+                         ~none=breakable_space,
+                         ~lead=space,
+                         ~trail=breakable_space,
+                         prev.pexp_loc,
+                         next.loc,
+                       ),
+                ~trail=
+                  ((_, prev)) =>
+                    print_comment_range(
+                      ~lead=space,
+                      ~block_end=true,
+                      prev.pexp_loc,
+                      enclosing_end_location(expr.pexp_loc),
+                    ),
+                ~f=print_punnable_expression,
+                exprs,
+              ),
+            )
+          }
+        )
+      | PExpBlock(exprs) =>
+        block_braces(
+          concat_map(
+            ~lead=
+              first =>
+                print_comment_range(
+                  ~block_start=true,
+                  ~trail=space,
+                  enclosing_start_location(expr.pexp_loc),
+                  first.pexp_loc,
+                ),
+            ~sep=
+              (prev, next) =>
+                print_comment_range(
+                  ~none=
+                    switch (
+                      next.pexp_loc.loc_start.pos_lnum
+                      - prev.pexp_loc.loc_end.pos_lnum
+                    ) {
+                    | 0
+                    | 1 => hardline
+                    | _ => hardline ++ hardline
+                    },
+                  ~lead=space,
+                  ~trail=space,
+                  prev.pexp_loc,
+                  next.pexp_loc,
+                ),
+            ~trail=
+              last =>
+                print_comment_range(
+                  ~block_end=true,
+                  ~lead=space,
+                  last.pexp_loc,
+                  enclosing_end_location(expr.pexp_loc),
+                ),
+            ~f=
+              e =>
+                if (has_disable_formatting_comment(e.pexp_loc)) {
```

Comment:
Shouldn't this be done everywhere? It seems to only be used a handful of times but I'd think you'd be able to disable formatting on anything.

Reply:
#2040